daemon/daemon_unix.go

```diff
@@ -159,6 +159,21 @@ func getBlkioWeightDevices(config containertypes.Resources) ([]specs.WeightDevic
 	return blkioWeightDevices, nil
 }
 
+func parseAndSetSecurityOpt(daemon *Daemon, container *container.Container, config *containertypes.HostConfig) error {
+	var findNNP bool
+	err := parseSecurityOpt(container, config)
+	for _, opt := range config.SecurityOpt {
+		if strings.Contains(opt, "no-new-privileges") {
```

Comment:
should not match : `really-no-new-privileges`, `no-new-privileges-please=true` etc